src/compiler/transformer.ts

```diff
@@ -100,9 +100,12 @@ namespace ts {
         return transformers;
     }
 
-    function getDeclarationTransformers(customTransformers?: CustomTransformers) {
+    function getDeclarationTransformers(compilerOptions: CompilerOptions, customTransformers?: CustomTransformers) {
         const transformers: TransformerFactory<SourceFile | Bundle>[] = [];
         transformers.push(transformDeclarations);
+        if (getEmitPrettyDeclarations(compilerOptions)) {

```

Comment:
This isnt checking if this transform is for incremental signature purposes but i think its a good idea to do it this way.. We try to update signatures to d.ts emit hash during emit so not having discrepancies between d.ts emit for incremental signature purpose and emit is better..